ivy/functional/backends/paddle/elementwise.py

```diff
@@ -830,39 +830,45 @@ def pow(
     return paddle.pow(x1, x2)
 
 
-def round(
-    x: paddle.Tensor, /, *, decimals: int = 0, out: Optional[paddle.Tensor] = None
-) -> paddle.Tensor:
-    def _np_round(x):
-        # this is a logic to mimic np.round behaviour
-        # which rounds odd numbers up and even numbers down at limits like 0.5
+# Implementation based on TensorFlow's scalar_round_half_to_even_op logic
+# Reference: https://github.com/tensorflow/tensorflow/blob/7f1050a6976d11bfb0bb37bdfc82350c0a238faa/tensorflow/core/kernels/cwise_ops.h#L510  # noqa: E501
+def _round_half_to_even(x):
+    round_val = paddle.floor(x + 0.5)
+    fraction = round_val - x
 
-        one = paddle.to_tensor(1, dtype="int64")
+    # Identify elements with a fractional part of 0.5
+    mask = paddle.equal(fraction, paddle.to_tensor(0.5, dtype=fraction.dtype))
 
-        # check if the number is even or odd
-        is_even = paddle.bitwise_and(paddle_backend.trunc(x).astype("int64"), one) == 0
+    # Round to the nearest even number if the fraction is 0.5
+    even_round_val = 2 * paddle.floor(0.5 * x + 0.5)
 
-        # round the number to the nearest integer
-        round_x = paddle.sign(x) * paddle.where(
-            is_even, paddle.floor(x.abs()), paddle.ceil(x.abs())
-        )
+    # Combine the results
+    return paddle.where(mask, even_round_val, round_val)
 
-        # if the number was rounded up from an even number
-        #   round the number down to the nearest even number
-        return paddle.where(
-            paddle.logical_and(
-                paddle.bitwise_and(round_x.astype("int64"), one) == 1.0,
-                is_even,
-            ),
-            round_x - 1.0,
-            round_x,
-        )
 
-    if x.dtype not in [paddle.float32, paddle.float64]:
-        if paddle.is_complex(x):
-            return paddle.complex(_np_round(x.real()), _np_round(x.imag()))
-        return _np_round(x.astype("float32")).astype(x.dtype)
-    return _np_round(x).astype(x.dtype)
+# This function aims to mimic the behavior of np.round similar to how tf.experimental.numpy.round does # noqa: E501
+# Reference for tf.experimental.numpy.round:https://github.com/tensorflow/tensorflow/blob/v2.13.0/tensorflow/python/ops/numpy_ops/np_array_ops.py#L724 # noqa: E501
+def round(
+    x: paddle.Tensor, /, *, decimals: int = 0, out: Optional[paddle.Tensor] = None
+) -> paddle.Tensor:
+    x = paddle.to_tensor(x, dtype=x.dtype)
+    dtype_ = x.dtype
+    factor = math.pow(10, decimals)
+
+    # Handle floating point and complex numbers
+    if paddle.is_floating_point(x) or paddle.is_complex(x):
+        factor = paddle.to_tensor(factor)
+        factor = paddle.cast(factor, dtype_)
+    else:
+        float_dtype_ = paddle.float32  # paddle.get_default_dtype()
+        x = x.astype(float_dtype_)
+        factor = paddle.cast(factor, float_dtype_)
+
+    x = paddle.multiply(x, factor)
+    x = _round_half_to_even(x)
+    x = paddle.divide(x, factor)
+
+    return x.astype(dtype_)
 
```

Comment:
The frontend changes are great. Ran the test locally and it passes now. 
But `ivy.round`'s backend is still failing.  I'm getting `ivy.utils.exceptions.IvyValueError: paddle: round: (InvalidArgument) cast(): argument 'x' (position 0) must be Tensor, but got float`.

Reply:
> The frontend changes are great. Ran the test locally and it passes now. But `ivy.round`'s backend is still failing. I'm getting `ivy.utils.exceptions.IvyValueError: paddle: round: (InvalidArgument) cast(): argument 'x' (position 0) must be Tensor, but got float`.

Let me take a look, it might due to `factor`